code/Spikes/Class_WiFiService/WiFiService.cpp

```diff
@@ -0,0 +1,131 @@
+
+#include <Arduino.h>
+#include "WiFiService.h"
+#include <SoftwareSerial.h>
+
+
+// Create SoftwareSerial object
+SoftwareSerial mySerial(10, 11); // RX, TX
+
+//Constructor
+WiFiService::WiFiService()
+{
+  StringComplete = false;
+  StringStarted = false;
+  rxString.reserve(50);  
+  rxString = "Test";
+  SerialChar = "";
+  StringLength = 0;
+  mySerial.begin(9600);   // Set data rate for the SW serial port
+  while (!mySerial) {;}   // wait for SW serial port to connect.
+}
+
+void WiFiService::Init(bool mode)
+{
+  //Mode "Wait_for_Start_Character": Does Bidirectional communication until it sees '['-Character, then Mode "BuildString" is entered in teh next Main Loop.
+  if (mode == true && !StringComplete)        {Wait_for_Start_Character();  HWtoSWSerial();}  
+
+  //Mode "BuildString": SWtoHW Communiction is inhibited and Characters are stored until ']' is seen and SWtoHW Communication is back active.
+  //The stored String can now be provided via Get_String()
+  if (mode == true && StringStarted == true)  {BuildString();               HWtoSWSerial();} 
+  
+  //Mode "Bidirectional". Just bidirectional Mode between SW and HW Serial.
+  if (mode == false || StringComplete)        {Bidirectional_Mode();        HWtoSWSerial();}
+  
+}
+
+
+// Checks Byte by Byte the SW Serial Buffer for the Start Character. 
+// When detected, StringInProgress = true and rxString = "["
+
+void WiFiService::Wait_for_Start_Character()
+{
+  while (mySerial.available() && !StringStarted) 
+  {
+    SerialChar = (char)mySerial.peek();
+    if (SerialChar == '[')
```

Comment:
Ich stimme zu (mit den Start/Stopp-Zeichen). Wir koennen das auch gerne machen und muessen uns nur entscheiden. Fuer das Android Team scheint es egal zu sein! Auch dien Punkt mit dem ausblenden von den Wifly standard strings stimmt. Lass uns die Diskussion in Slack anstoessen und endgueltig entscheiden ob wir '[' und ']' als Terminierungszeichen wollen.